File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        arr=[]
        res=[]
        obj1=defaultdict(bool)
        for i in range(len(points)):
            s=points[i][0]**2+points[i][1]**2
            if not obj1[s]:
                obj1[s]=[points[i]]
            else:
                obj1[s].append(points[i])
            arr.append(s)
        arr.sort()
        
        for i in range(k):
            res.append(obj1[arr[i]].pop())
        return res

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.js
/**
 * @param {number[][]} points
 * @param {number} k
 * @return {number[][]}
 */
var kClosest = function(points, k) {
    points.sort(numCompare)
   return points.slice(0,k)
};
function numCompare(num1,num2){
    return (num1[0]**2+num1[1]**2)-(num2[0]**2+num2[1]**2)
}